[ImportVerilog] ltl.implication op types don't verify with rose in antecendent

This seems to be a MWE:

```sv
module simple (
    input logic in1, in2
);

    always assert property ($rose(in1) |-> in2);
endmodule
```

This throws the error: 

```
simple.sv:5:12: error: 'ltl.implication' op operand #0 must be 1-bit signless integer or LTL sequence type, but got '!ltl.property'
    always assert property ($rose(in1) |-> in2);
           ^
simple.sv:5:12: note: see current operation: %9 = "ltl.implication"(%6, %8) : (!ltl.property, i1) -> !ltl.property
```

I guess the rose construct is generating some invalid input type for the ltl.implication op that eventually gets generated

--- a/lib/Conversion/ImportVerilog/AssertionExpr.cpp
+++ b/lib/Conversion/ImportVerilog/AssertionExpr.cpp
@@ -8,6 +8,7 @@
 
 #include "slang/ast/expressions/AssertionExpr.h"
 #include "ImportVerilogInternals.h"
+#include "circt/Dialect/Comb/CombOps.h"
 #include "circt/Dialect/LTL/LTLOps.h"
 #include "circt/Dialect/Moore/MooreOps.h"
 #include "circt/Support/LLVM.h"
@@ -87,7 +88,8 @@ struct AssertionExprVisitor {
     auto valueType = value.getType();
     // For assertion instances the value is already the expected type, convert
     // boolean value
-    if (!mlir::isa<ltl::SequenceType, ltl::PropertyType>(valueType)) {
+    if (!mlir::isa<ltl::SequenceType, ltl::PropertyType, mlir::IntegerType>(
+            valueType)) {
       value = context.convertToI1(value);
     }
     if (!value)
@@ -316,8 +318,7 @@ FailureOr<Value> Context::convertAssertionSystemCallArity1(
                 [&]() -> Value {
                   auto past =
                       ltl::PastOp::create(builder, loc, value, 1).getResult();
-                  auto notPast =
-                      ltl::NotOp::create(builder, loc, past).getResult();
+                  auto notPast = comb::createOrFoldNot(loc, past, builder);
                   auto current = value;
                   auto notPastAndCurrent =
                       ltl::AndOp::create(builder, loc, {current, notPast})

--- a/test/Conversion/ImportVerilog/builtins.sv
+++ b/test/Conversion/ImportVerilog/builtins.sv
@@ -375,8 +375,9 @@ module SampleValueBuiltins #() (
   // CHECK-NEXT: [[C2:%.+]] = moore.read [[CLKWIRE]] : <l1>
   // CHECK-NEXT: [[CURRENT:%.+]] = moore.to_builtin_bool [[C2]] : l1
   // CHECK-NEXT: [[PAST:%.+]] = ltl.past [[CURRENT]], 1 : i1
-  // CHECK-NEXT: [[NOTPAST:%.+]] = ltl.not [[PAST]] : i1
-  // CHECK-NEXT: [[NOTPASTANDCURRENT:%.+]] = ltl.and [[CURRENT]], [[NOTPAST]] : i1, !ltl.property
+  // CHECK-NEXT: [[TRUE:%.+]] = hw.constant true
+  // CHECK-NEXT: [[NOTPAST:%.+]] = comb.xor [[PAST]], [[TRUE]] : i1
+  // CHECK-NEXT: [[NOTPASTANDCURRENT:%.+]] = ltl.and [[CURRENT]], [[NOTPAST]] : i1, i1
   rising_clk: assert property (@(posedge clk_i) clk_i |=> $rose(clk_i));
   // CHECK: moore.procedure always {
   // CHECK-NEXT: [[C:%.+]] = moore.read [[CLKWIRE]] : <l1>

